lib/wsdl.js

```diff
@@ -1932,10 +1932,22 @@ WSDL.prototype.findSchemaType = function(name, nsURI) {
   return schema.complexTypes[name];
 };
 
-WSDL.prototype.findChildSchemaObject = function(parameterTypeObj, childName) {
+WSDL.prototype.findChildSchemaObject = function(parameterTypeObj, childName, backtrace) {
   if (!parameterTypeObj || !childName) {
     return null;
   }
+
+  if (!backtrace) {
+    backtrace = [];
+  }
+
+  if (backtrace.indexOf(parameterTypeObj) >= 0) {
+    // We've recursed back to ourselves; break.
+    return null;
+  } else {
+    backtrace = backtrace.concat([parameterTypeObj]);
```

Comment:
Sure. Let's say we're traversing this schema:

```
foo
  bar
    foo
    baz
  blah
zub
```

If we used `.push()`, here's what the array would look like as we do the depth-first traversal:

```
[foo]
[foo, bar]
// (here we encounter foo again, so we skip)
[foo, bar, baz]
[foo, bar, baz, blah] // Incorrect; should be [foo, blah]
[foo, bar, baz, blah, zub] // Incorrect; should be [zub]
```